### PR TITLE
Add eslint to themes and classic folders

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | In order to upgrade the quality and consistency of this folder, as we did for admin themes and more, we want eslint inside it!
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | It's mainly refacto, so everything in the FO should work as expected

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20080)

### CODE REVIEW

**Comment by PierreRambaud** [comment]
Some tests are failing, watch out, babel is not present in the classic theme :wink:

**Comment by Progi1984** [comment]
@NeOMakinG Travis is :red_circle: & You have a conflicting file

**Review by Progi1984 — COMMENTED** [review]
Miss the Github Action for checking at each PR

File: themes/_core/js/facets.js [hunk 1]
```diff
@@ -23,52 +23,53 @@
...
+  window.history.pushState(data, document.title, data.current_url);
 }
 
 function handleError(xhr, textStatus) {
-    if (textStatus === 'abort') {
-        return;
-    }
-    // TODO: feedback
+  if (textStatus === 'abort') {
+    console.warn('Request aborted');
```
Comment (Progi1984) [inline themes/_core/js/facets.js]: Is it wanted the behavior change ?

Comment (NeOMakinG) [reply themes/_core/js/facets.js]: Changed it into the correct behavior, idk why my conflict was empty, so I decided to put another behavior, but thanks to you I seen the past one

File: themes/classic/_dev/js/product.js [hunk 2]
```diff
@@ -85,8 +85,10 @@ $(document).ready(function () {
 
   function createInputFile() {
     $(prestashop.themeSelectors.fileInput).on('change', (event) => {
-      let target, file;
+      let target; let
+        file;
```
Comment (Progi1984) [inline themes/classic/_dev/js/product.js]:
```suggestion
      let target;
      let file;
```

Normal ?

**Review by Progi1984 — COMMENTED** [review]
@NeOMakinG A last comment

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by sowbiba — DISMISSED** [review]
Great job. Thanks @NeOMakinG 
If I want to run it locally, do I need to use node 10.x ? or the 8.17 can do the job?
I suppose we have a npm run lint fix? (personal curiosity)

EDIT : You have some conflicts to handle

**Comment by NeOMakinG** [comment]
> Great job. Thanks @NeOMakinG
> If I want to run it locally, do I need to use node 10.x ? or the 8.17 can do the job?
> I suppose we have a npm run lint fix? (personal curiosity)
> 
> EDIT : You have some conflicts to handle

Just tested with 8.17 and it run perfectly, but I would suggest you to use at least node 12 as it's the current LTS

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

**Comment by Roomsmaniia** [comment]
hi @NeOMakinG,
i got this with the PR, but i have the good translation without the PR
image

**Comment by NeOMakinG** [comment]
> hi @NeOMakinG,
> i got this with the PR, but i have the good translation without the PR
> image

I can't be related to be honest, I'm only touching JS on this PR 🤕

**Comment by Roomsmaniia** [comment]
LGTM,

Thanks @NeOMakinG

**Comment by Progi1984** [comment]
Thanks @NeOMakinG & @RomainBocheux

**Pull request merged**